ydb/query/base.py

```diff
@@ -0,0 +1,332 @@
+        """
+        Explicitly begins a transaction
+
+        :param settings: A request settings
+
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def commit(settings: QueryClientSettings = None):
+        """
+        Calls commit on a transaction if it is open. If transaction execution
+        failed then this method raises PreconditionFailed.
+
+        :param settings: A request settings
+
+        :return: A committed transaction or exception if commit is failed
+        """
+        pass
+
+    @abc.abstractmethod
+    def rollback(settings: QueryClientSettings = None):
+        """
+        Calls rollback on a transaction if it is open. If transaction execution
+        failed then this method raises PreconditionFailed.
+
+        :param settings: A request settings
+
+        :return: A rolled back transaction or exception if rollback is failed
+        """
+        pass
+
+    @abc.abstractmethod
+    def execute(
+        self,
+        query: str,
+        commit_tx: bool = False,
+        tx_mode: BaseQueryTxMode = None,
+        syntax: QuerySyntax = None,
+        exec_mode: QueryExecMode = None,
+        parameters: dict = None,
+        concurrent_result_sets: bool = False,
+    ):
+        """
+        Sends a query to Query Service
+        :param query: (YQL or SQL text) to be executed.
+        :param commit_tx: A special flag that allows transaction commit.
+        :param tx_mode: Transaction mode, which is a one from the following choises:
+         1) QuerySerializableReadWrite() which is default mode;
+         2) QueryOnlineReadOnly(allow_inconsistent_reads=False);
+         3) QuerySnapshotReadOnly();
+         4) QueryStaleReadOnly().
+        :param syntax: Syntax of the query, which is a one from the following choises:
+         1) QuerySyntax.YQL_V1, which is default;
+         2) QuerySyntax.PG.
+        :param exec_mode: Exec mode of the query, which is a one from the following choises:
+         1) QueryExecMode.EXECUTE, which is default;
+         2) QueryExecMode.EXPLAIN;
+         3) QueryExecMode.VALIDATE;
+         4) QueryExecMode.PARSE.
+        :param parameters: dict with parameters and YDB types;
+        :param concurrent_result_sets: A flag to allow YDB mix parts of different result sets. Default is False;
+
+        :return: Iterator with result sets
+        """
+        pass
+
+
+class IQueryClient(abc.ABC):
+    def __init__(self, driver: SupportedDriverType, query_client_settings: QueryClientSettings = None):
+        pass
+
+    @abc.abstractmethod
+    def session(self) -> IQuerySession:
+        pass
+
+
+def create_execute_query_request(
+    query: str,
+    session_id: str,
+    tx_id: str = None,
+    commit_tx: bool = False,
+    tx_mode: BaseQueryTxMode = None,
+    syntax: QuerySyntax = None,
+    exec_mode: QueryExecMode = None,
+    parameters: dict = None,
+    concurrent_result_sets: bool = False,
+    empty_tx_control: bool = False,
```

Comment:
Why we need separate flag empty_tx_control? What is use tx_mode = None for it?

And use SerializeReadWrite (or add some constant) as default value for methods instead of None?